src/markerStyle.ts

```diff
@@ -0,0 +1,156 @@
+import * as vscode from 'vscode'
+
+interface Config {
+  markerStyle?: string
+  markerWholeLine?: boolean
+  markerColor?: string
+  markerJSON?: MyDecorationRenderOptions // will auto generate
+  markerJSONCustom?: MyDecorationRenderOptions
+}
+interface Edit {
+  line: number
+  character: number
+  filepath: string
+  filename: string
+  lineText: string
+}
+
+let globalConfig: Config = {}
+let context: vscode.ExtensionContext
+let editList: Edit[]
+let myDecoration: vscode.TextEditorDecorationType | undefined = undefined
+
+export const reloadStyleConfig = () => {
+  //console.log('reloadStyleConfig')
+  const config = vscode.workspace.getConfiguration('navigateEditHistory')
+  const newConfig: Config = {
+    markerStyle: config.get<string>('markerStyle'),
+    markerWholeLine: config.get<boolean>('markerWholeLine'),
+    markerColor: config.get('markerColor'),
+    markerJSONCustom: config.get('markerJSONCustom'),
+  }
+  newConfig.markerJSON = generateJSON(newConfig)
+  globalConfig = newConfig
+  myDecoration = undefined // need init myDecoration
+  //console.log('globalConfig:'+ JSON.stringify(globalConfig) );
+}
+function generateJSON(conf: Config): MyDecorationRenderOptions {
+  //  show in overview 默认在右侧缩略图中显示
+  let objDecoration: MyDecorationRenderOptions = {
+    overviewRulerColor: '#AA000099', // RGBA
+    overviewRulerLane: 2, // Center: 2 Full: 7 Left: 1 Right: 4
```

Comment:
Unclear what these numbers mean.

Reply:
those number is from VSCode reference , https://code.visualstudio.com/api/references/vscode-api#OverviewRulerLane 
set the postion on overview( right scroller )